### PR TITLE
Add the scroll to top feature for the website

### LINKED ISSUE #47: Add a "scroll to top" button in the page
As the links in the navbar take you to different sections of the page, you might often feel the need to scroll to the top of the page. Adding a "scroll to top" button on the page would help users a lot, as the portfolio website is getting pretty long now.

### PR DESCRIPTION
**Pull request by prakharrathi25**
### Why This PR Adds Value

As the links in the navbar take you to different sections of the page, you might have to scroll to the top of the page. Adding a "scroll to top" button on the page would help users as this website keeps getting longer. 

### What This PR Adds

This PR adds a button on the bottom right which takes you to the top of the page when clicked. The button only appears after you scroll down a bit and not immediately from the start. 

#### Files Added 
- `footer.html`: This is where the button resides and I have also added a script to implement the scroll to top feature. 
- `footer.scss`: File created to style the button. 

#### Files changes 
- `index.html`: Added the footer to the page
- `all.scss`: Added the footer sass import 

### Screenshot

image

### How This PR Could Be Improved

- **Improvement Area** - The scroll speed could be changed and the button design could be improved. 

### Issue This PR Closes

This closes #47

### CODE REVIEW

File: _sass/footer.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    background-color: #333;
+  }
+  #btnTop:active {
+    background-color: #555;
+  }
+  #btnTop.show {
+    opacity: 1;
+    visibility: visible;
+  }
+
```
Comment (debdutgoswami) [inline _sass/footer.scss:30]: To get the smooth scrolling effect, just add the below css and it should do the trick.

```css
html {
    scroll-behavior: smooth;
}
```

**Review by debdutgoswami — CHANGES_REQUESTED** [review]

File: _sass/footer.scss (hunk 1 above)
Comment (prakharrathi25) [reply _sass/footer.scss:30]: Wow, this was quite a simple solution. Everyone online keeps posting Jqueries for this. Also, is there a way to set the scroll speed? That would be an additional thing but would be quite helpful. Thanks a lot!

Comment (debdutgoswami) [reply _sass/footer.scss:30]: I think jQuery would be needed to set the scroll speed, although I am not really sure if any css hack is present or not.

@louisefindlay23 do you think there is any css hack for scroll speed?

**Review by debdutgoswami — APPROVED** [review]
You can move the main `html` styling to `all.scss` since it is a global style. It is subjective though. Maybe I am being a bit nitpicking. Let's see what others have to say.

Other than this minor subjective refactoring, this LGTM

**Review by robzwolf — CHANGES_REQUESTED** [review]
Love to see cool stuff like this, especially when solved with simple CSS!

Can you fix the merge conflict and then re-request a review from me?

**Comment by prakharrathi25** [comment]
> Love to see cool stuff like this, especially when solved with simple CSS!
> 
> Can you fix the merge conflict and then re-request a review from me?

Thanks, fixed the conflict!

**Comment by prakharrathi25** [comment]
@robzwolf can you check out this PR? I sorted out the conflicts twice but other PRs keep introducing new conflicts haha! Thanks!

File: _sass/footer.scss (hunk 1 above)
Comment (louisefindlay23) [reply _sass/footer.scss:30]: Done some research. There are ways but they seem pretty complex.

This [vanilla JS snippet](https://www.sololearn.com/Discuss/1748649/how-to-change-scroll-speed) might work so you don't need the overhead of jQuery (if we're not already using it - not sure).

Not tested it though so YMMV. Hope it helps.

**Review by robzwolf — APPROVED** [review]

**Pull request merged**